toolkit/statemachine/src/androidTest/kotlin/br/com/arch/toolkit/statemachine/SceneStateMachineTest.kt

```diff
@@ -0,0 +1,89 @@
+package br.com.arch.toolkit.statemachine
+
+import android.content.Intent
+import android.support.test.espresso.intent.rule.IntentsTestRule
+import android.support.test.runner.AndroidJUnit4
+import android.transition.Fade
+import android.transition.Scene
+import android.widget.FrameLayout
+import br.com.arch.toolkit.test.TestActivity
+import br.com.concretesolutions.kappuccino.utils.doWait
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class SceneStateMachineTest {
+
+    @Rule
+    @JvmField
+    val rule = IntentsTestRule(TestActivity::class.java, true, false)
+
+    @Test
+    fun whenChangeState_shouldApplyViewChangesOnEachState() {
+        lateinit var container: FrameLayout
+        rule.launchActivity(Intent())
+        rule.activity.setView {
+            container = FrameLayout(it)
+            container
+        }
+        doWait()
+
+        val machine = SceneStateMachine()
+        machine.setup {
+            state(0) {
+                scene(android.R.layout.activity_list_item to container)
+                onEnter { }
+            }
+            state(1) {
+                scene(android.R.layout.browser_link_context_header to container)
+                transition(Fade())
+                onExit { }
+            }
+            state(3) {
+                scene(Scene(container))
+            }
+        }
+
+        rule.activity.runOnUiThread {
+            machine.changeState(0)
+        }
+        doWait(1000L)
+
+        Assert.assertEquals(0, machine.currentStateKey)
+        Assert.assertNotNull(rule.activity.findViewById(android.R.id.icon))
```

Comment:
Because I wanted to test this class with robolectric, but I couldn't find any way to work with it.

So I kept that way thinking in move this class later.